tycho-indexer/src/substreams/stream.rs

```diff
@@ -103,6 +104,14 @@ fn stream_blocks(
                     for await response in stream {
                         match process_substreams_response(response).await {
                             BlockProcessedResult::BlockScopedData(block_scoped_data) => {
+                                if let Some(block) = block_scoped_data.clock.clone() {
+                                    if let Some(block_ts) = block.timestamp {
+                                        let now = SystemTime::now().duration_since(UNIX_EPOCH).expect("Time went backwards!?").as_secs();
+                                        let lag = max(now - block_ts.seconds as u64, 0);
+                                        gauge!("substreams_lag_seconds", "extractor" => extractor_id.clone()).set(lag as f64);
+                                    }
+                                };
+
```

Comment:
Yeah much simpler! A bit surprised that these fields are optional though. Any idea why?

Reply:
No clue - I ran this a few times and had a test metric to log whenever it was None and it was never logged. It seems like in general we can expect a value here.